tests/integration/queryparam.spec.ts

```diff
@@ -0,0 +1,152 @@
+        let dino = new Dino(app, '/api');
+
+        dino.registerController(TestControllerFake);
+        dino.useRouter(() => express.Router());
+        dino.bind();
+
+        request(app)
+            .get('/api/testcontroller/toBoolean?val=true')
+            .expect('Content-Type', /json/)
+            .expect(200, { data: true });
+        request(app)
+            .get('/api/testcontroller/toBoolean?val=True')
+            .expect('Content-Type', /json/)
+            .expect(200, { data: true });
+        request(app)
+            .get('/api/testcontroller/toBoolean?val=TRUE')
+            .expect('Content-Type', /json/)
+            .expect(200, { data: true });
+        request(app)
+            .get('/api/testcontroller/toBoolean?val=false')
+            .expect('Content-Type', /json/)
+            .expect(200, { data: false });
+        request(app)
+            .get('/api/testcontroller/toBoolean?val=false')
+            .expect('Content-Type', /json/)
+            .expect(200, { data: false });
+        request(app)
+            .get('/api/testcontroller/toBoolean?val=FALSE')
+            .expect('Content-Type', /json/)
+            .expect(200, { data: false }, done);
+    });
+
+    it('queryParam.toInteger.parses_integer', done => {
+        let app = express();
+        let dino = new Dino(app, '/api');
+
+        dino.registerController(TestControllerFake);
+        dino.useRouter(() => express.Router());
+        dino.bind();
+
+        request(app)
+            .get('/api/testcontroller/toInteger?val=5')
+            .expect('Content-Type', /json/)
+            .expect(200, { data: 5 });
+        request(app)
+            .get('/api/testcontroller/toInteger?val=5')
+            .expect('Content-Type', /json/)
+            .expect(200, { data: 0 });
+        request(app)
+            .get('/api/testcontroller/toInteger?val=5.5')
+            .expect('Content-Type', /json/)
+            .expect(400);
+        request(app)
+            .get('/api/testcontroller/toInteger?val=-5')
+            .expect('Content-Type', /json/)
+            .expect(200, { data: -5 }, done);
+    });
+
+    it('queryParam.toNumber.parses_number', done => {
+        let app = express();
+        let dino = new Dino(app, '/api');
+
+        dino.registerController(TestControllerFake);
+        dino.useRouter(() => express.Router());
+        dino.bind();
+
+        request(app)
+            .get('/api/testcontroller/toNumber?val=5.5')
+            .expect('Content-Type', /json/)
+            .expect(200, { data: 5.5 });
+        request(app)
+            .get('/api/testcontroller/toNumber?val=0')
+            .expect('Content-Type', /json/)
+            .expect(200, { data: 0 });
+        request(app)
+            .get('/api/testcontroller/toNumber?val=-5.5')
+            .expect('Content-Type', /json/)
+            .expect(200, { data: -5.5 }, done);
+    });
+
+    it('queryParam.overrides_pathParam_with_same_keyName', done => {
```

Comment:
Pass in vals for pathParam and queryParam with same controller param name and verify the queryParam is the one selected.